### PR TITLE
feat: add more metrics for compactor

### PR DESCRIPTION
**Pull request by NGA-TRAN**
I want a few more metrics for our [Compactor Dashboard](https://grafana.prod01-eu-central-1.aws.influxdata.io/d/z7-FhLsnk/compactor?orgId=1)

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: compactor/src/compact.rs [hunk 1]
```diff
@@ -424,14 +446,18 @@ impl Compactor {
...
 
-            // Only one file without tombstones, no need to compact, upgrade it since it is
-            // non-overlapping
+            // Only one file without tombstones, no need to compact.
             if group.parquet_files.len() == 1 && group.tombstones.is_empty() {
-                upgrade_level_list.push(group.parquet_files[0].parquet_file_id());
+                actual_compacted_file_count -= 1;
+                // If it is level 0, upgrade it since it is non-overlapping
+                if group.parquet_files[0].data.compaction_level == INITIAL_COMPACTION_LEVEL {
+                    upgrade_level_list.push(group.parquet_files[0].parquet_file_id());
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:459]: No need to upgrade if it is already level 1

File: compactor/src/compact.rs [hunk 2]
```diff
@@ -237,6 +245,17 @@ impl Compactor {
             "compactor_compacted_files_total",
             "counter for the number of files compacted",
         );
+
+        let compaction_actual_counter = registry.register_metric(
```
Comment (alamb) [inline compactor/src/compact.rs:249]: I wonder if calling this "compactor_input_compacted_files_total" would be clearer?

**Review by alamb — APPROVED** [review]
Is there some way to write tests for this code?

**Comment by NGA-TRAN** [comment]
@alamb 
> Is there some way to write tests for this code?
All the available compactor counters do not have tests. When I spend more time on this, I will see how we tests similar metrics and add them in

**Pull request merged**